File: My_Notes/Programming/Samples/async_samples/C#/Async/Async/Async2.cs
using System.Threading.Tasks;

namespace Async
{
    partial class Program
    {
        static async void Async2()
        {
            var job = new Job(3, 1, 2);
            var dbEngineer = new DBEngineer();
            var apiEngineer = new APIEngineer();
            var uiEngineer = new UIEngineer();
            Task task1 = dbEngineer.DoAsync(job);
            Task task3 = uiEngineer.DoAsync(job);

            await task1;
            Task task2 = apiEngineer.DoAsync(job);
            await Task.WhenAll(task2, task3);
        }
    }
}

File: My_Notes/Programming/Samples/async_samples/dart/async2.dart
import 'dart:io';
import './job.dart';
import './engineer.dart';

void async2(){
  Job job = Job(3, 1, 2);
  DBEngineer dbEngineer = DBEngineer();
  APIEngineer apiEngineer = APIEngineer();
  UIEngineer uiEngineer = UIEngineer();
  Future<void> task1 = dbEngineer.doTaskAsync(job).then((value) => apiEngineer.doTaskAsync(job));
  Future<void> task2 = uiEngineer.doTaskAsync(job);

  Future.wait({task1, task2});
}